src/include/migraphx/op/quantizelinear.hpp

```diff
@@ -40,7 +40,11 @@ struct quantizelinear
     std::string name() const { return "quantizelinear"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.same_dims();
+        check_shapes{inputs, *this}.same_dims().has(2, 3);
+        if(inputs[0].type() != inputs[1].type())
```

Comment:
What do you mean? Zero point needs to be the same type as the input.

Reply:
also `onnxspec` doesn't say that `scales` and `input` type need to match. It says `scale` is `float`. and input can be either int32 or float. We may need to update onnx parser if we want both of them to match.

Reply:
> if there is zero_point argument, it needs to be uint8 type.

It can be int8 or uint8, since the output can be int8 or unint8, and the output and zero_point need to be the same type.

Ideally, i dont think we need to enforce it to be 8-bit integers as we could use this for other quantization types in the future.

> We may need to update onnx parser if we want both of them to match.

Its already updated in this PR to insert the converts by using `add_common_args`. We have onnx tests that use int32 input.